examples/widget_impl.rs

```diff
@@ -0,0 +1,342 @@
+//! # [Ratatui] Widgets implementation examples
+//!
+//! This example demonstrates various ways to implement widget traits in Ratatui on a type, a
+//! reference, and a mutable reference. It also shows how to use the `WidgetRef` trait to render
+//! boxed widgets.
+//!
+//! The latest version of this example is available in the [examples] folder in the repository.
+//!
+//! Please note that the examples are designed to be run against the `main` branch of the Github
+//! repository. This means that you may not be able to compile with the latest release version on
+//! crates.io, or the one that you have installed locally.
+//!
+//! See the [examples readme] for more information on finding examples that match the version of the
+//! library you are using.
+//!
+//! [Ratatui]: https://github.com/ratatui-org/ratatui
+//! [examples]: https://github.com/ratatui-org/ratatui/blob/main/examples
+//! [examples readme]: https://github.com/ratatui-org/ratatui/blob/main/examples/README.md
+use std::{
+    iter,
+    time::{Duration, Instant},
+};
+
+use color_eyre::Result;
+use common::Terminal;
+use crossterm::event::{self, Event, KeyCode};
+use ratatui::{
+    buffer::Buffer,
+    layout::{Constraint, Layout, Position, Rect, Size},
+    style::{Color, Style},
+    widgets::{Widget, WidgetRef},
+};
+
+fn main() -> Result<()> {
+    common::install_hooks()?;
+    let tui = common::init_terminal()?;
+    let app = App::default();
+    app.run(tui)?;
+    common::restore_terminal()?;
+    Ok(())
+}
+
+#[derive(Default)]
+struct App {
+    should_quit: bool,
+    timer: Timer,
+    #[cfg(feature = "unstable-widget-ref")]
```

Comment:
This is already a required feature for this example.

```suggestion
```

(there are multiple cases of this in this PR)

Reply:
This is an intentional prophylactic against the regular amount "Why doesn't this compile" type questions from newer rust users.

There are multiple paths to get to seeing the examples. Newer users may not be aware of where this is set in the cargo.toml. They may be reading this on github without the context, or have copied the example code into their own projects.